packages/flutter_tools/lib/src/plugins.dart

```diff
@@ -316,21 +352,41 @@ Future<void> _writeAndroidPluginRegistrant(FlutterProject project, List<Plugin>
   final Map<String, dynamic> context = <String, dynamic>{
     'plugins': androidPlugins,
   };
-
   final String javaSourcePath = fs.path.join(
     project.android.pluginRegistrantHost.path,
     'src',
     'main',
     'java',
   );
+
+  bool canUseNewEmbedding = false;
+  if (featureFlags.isNewAndroidEmbeddingEnabled) {
+    assert(project.android != null);
+    final File androidManifest = project.android.appManifestFile;
+    assert(androidManifest.existsSync());
+    final String manifestContent = androidManifest.readAsStringSync();
+    if (!manifestContent.contains('usesNewEmbedding:true')) {
```

Comment:
`usesNewEmbedding` needs to be in the Android manifest? I think this is the first I've seen of this requirement. I think we should avoid marking up the Android manifest for the embedding. Also, any permanent requirement that refers to `new` will be a problem in the future.

Reply:
I'm open for suggestions. Maybe `embedding:2.0` ?  Should the new embedding have a name?